### PR TITLE
Character counter tests

### PR DESCRIPTION
**Pull request by johnbender**
@zachleat 

Questions for you posted inline. Please double check the refactor.

### CODE REVIEW

File: js/character-counter.js [hunk 1]
```diff
@@ -3,44 +3,68 @@
...
+		} else {
+			add.push( "singular" );
+			remove.push( "plural" );
 		}
+
+		var hitMax = this.max && newval === 0;
+		var hitMin = this.min && newval <= 0;
+
+		if( hitMax || hitMin ){
+			add.push( "limit " + ( hitMax ? "max" : "min" ) );
```
Comment (johnbender) [inline js/character-counter.js]: I'm confused here, it looks like `limit min` is added when the length of the value is larger than the minimum length. In that regard `hitMin` makes sense. Are those classes intended to hide any text alerting the user that they haven't hit the minimum? (Just double checking).

File: js/character-counter.js [hunk 2]
```diff
@@ -3,44 +3,68 @@
...
+		this.max = parseFloat( this.$el.attr( "maxlength" ), 10 );
+		this.min = parseFloat( this.$el.attr( "minlength" ), 10 );
+
+		if ( this.$label.length ){
+			this.$el
+				.unbind( ".charcount")
+				.bind( "input.charcount keyup.charcount", function(){
+					self.characterCount();
+				});
+		}
```
Comment (johnbender) [inline js/character-counter.js]: Note that I've moved a whole bunch of the `characterCount` method stuff to the constructor meaning it gets done once and only once. I can't see that being an issue but it's worth a double check.

File: js/character-counter.js (hunk 1 above)
Comment (zachleat) [inline js/character-counter.js]: Correct. There is a demo here: http://formcore.dev/views/text-inputs.html “EXAMPLE: MULTI-LINE TEXT, WITH CHARACTER COUNTER MINLENGTH”

**Comment by zachleat** [comment]
LGTM!

**Pull request merged**